src/mail-app/mail/model/ConversationListModel.ts

```diff
@@ -0,0 +1,555 @@
+				if (!conversation) {
+					return
+				}
+
+				await this.listModel.deleteLoadedItem(this.listModelIdOfConversation(conversation))
+				this.deleteSingleMail(mailId)
+
+				// The state changed in the meantime. We should not touch the conversation map.
+				if (this.conversationMap.get(conversation.conversationId)?.latestMail != conversation.latestMail) {
+					return
+				}
+				this.conversationMap.delete(conversation.conversationId)
+
+				// If there is an earlier mail in the conversation that is in this mail set, we want it to take this one's place.
+				let latestMail: LoadedMail | null = null
+				for (const mail of this.mailMap.values()) {
+					if (isSameId(listIdPart(mail.mail.conversationEntry), conversation.conversationId)) {
+						if (latestMail == null || firstBiggerThanSecondCustomId(elementIdPart(mail.mailSetEntryId), elementIdPart(latestMail.mailSetEntryId))) {
+							// Important: Map iterates in insertion order, which does not correspond to list sorting. As
+							// such, we can't break here, as there may be even newer mails after this.
+							latestMail = mail
+						}
+					}
+				}
+
+				if (latestMail) {
+					const newConversation = {
+						...conversation,
+						latestMail: elementIdPart(latestMail.mailSetEntryId),
+					}
+					this.conversationMap.set(conversation.conversationId, newConversation)
+					this.listModel.waitLoad(() => {
+						if (this.conversationMap.get(conversation.conversationId) === newConversation && this.listModel.canInsertItem(newConversation)) {
+							this.listModel.insertLoadedItem(newConversation)
+						}
+					})
+				}
+			} else if (update.operation === OperationType.CREATE) {
+				const loadedMail = await this.loadSingleMail([update.instanceListId, update.instanceId])
+				const addedMail = loadedMail.addedItems[0]
+				if (addedMail != null) {
+					await this.listModel.waitLoad(async () => {
+						if (this.listModel.canInsertItem(addedMail)) {
+							this.listModel.insertLoadedItem(addedMail)
+						}
+						for (const oldEntry of loadedMail.deletedItems) {
+							const id = this.listModelIdOfConversation(oldEntry)
+							const wasSelected = this.listModel.isItemSelected(id)
+							const inMultiselect = this.isInMultiselect()
+							const selection = this.listModel.getSelectedAsArray()
+
+							// ensure the selection does not change
+							if (wasSelected) {
+								if (inMultiselect) {
+									for (const s of selection) {
+										if (this.listModelIdOfConversation(s) !== this.listModelIdOfConversation(oldEntry)) {
+											this.listModel.onSingleInclusiveSelection(s)
+										}
+									}
+									this.listModel.onSingleInclusiveSelection(addedMail)
+								} else {
+									this.listModel.onSingleSelection(addedMail)
+								}
+							}
+
+							await this.listModel.deleteLoadedItem(id)
+						}
+					})
+				}
+			}
+		} else if (isUpdateForTypeRef(MailTypeRef, update)) {
+			// We only need to handle updates for Mail.
```

Comment:
because of entity event optimization we might also need to handle create, at least if we have it in the list.
see https://github.com/tutao/tutanota/pull/8473